### PR TITLE
added model mapper module

### PR DESCRIPTION
**Pull request by AlexKazmirchuk**
added own interface for model mapper , it implementation and spring configuration

### CODE REVIEW

**Comment by Dominys** [comment]
@AlexKazmirchuk Where is tests ?

**Comment by AlexKazmirchuk** [comment]
@Dominys Added tests

File: softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    @Before
+    public void init(){
+        Address address = new Address("Chicago","Galagers 43", "USA" );
+        person1 = new Person("Bob", "Fisher", 34, address);
+        person2 = new Person("John", null, 24, null);
+    }
+
+    @Test
+    public void fromPerson1ToPersonDTOmapMethodTest(){
```
Comment (Dominys) [inline softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java]: @AlexKazmirchuk Split into separate methods

File: softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        assertThat(mappedPerson.getName(), is(personDTO.getName()));
+        assertThat(mappedPerson.getLastName(), is(personDTO.getLastName()));
+        assertThat(mappedPerson.getAge(), is(personDTO.getAge()));
+        assertThat(mappedPerson.getAddress().getCity(), is(personDTO.getCity()));
+        assertThat(mappedPerson.getAddress().getStreet(), is(personDTO.getStreet()));
+        assertThat(mappedPerson.getAddress().getCountry(), is(personDTO.getCountry()));
+    }
+
+    @Test
+    public void fromPerson2ToPersonDTOmapMethodTest(){
```
Comment (Dominys) [inline softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java]: @AlexKazmirchuk same

File: softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+        assertThat(mappedPerson, notNullValue());
+        assertThat(mappedPerson.getName(), is(personDTO.getName()));
+        assertThat(mappedPerson.getLastName(), nullValue());
+        assertThat(mappedPerson.getAge(), is(personDTO.getAge()));
+        assertThat(mappedPerson.getAddress(), nullValue());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void mapWithWrongArgumentsTest1(){
```
Comment (Dominys) [inline softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java]: @AlexKazmirchuk Use proper names. 1, 2 etc. is not proper names as they did not provide any information.

File: softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/models/Address.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package com.softgroup.common.model.mapper.models;
+
+/**
+ * @author AlexKazmirchuk
+ * @since 31.03.17.
+ */
+public class Address {
```
Comment (Dominys) [inline softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/models/Address.java]: @AlexKazmirchuk Add test prefix for test models.

File: softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java (hunk 1 above)
Comment (AlexKazmirchuk) [reply softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java]: @Dominys splited

File: softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java (hunk 2 above)
Comment (AlexKazmirchuk) [reply softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java]: @Dominys splited

File: softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java (hunk 3 above)
Comment (AlexKazmirchuk) [reply softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/ModelMapperTest.java]: @Dominys renamed

File: softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/models/Address.java (hunk 4 above)
Comment (AlexKazmirchuk) [reply softgroup/basis/common-model-mapper/src/test/java/com/softgroup/common/model/mapper/models/Address.java]: @Dominys Added prefix to test models

**Review by Dominys — APPROVED** [review]

**Pull request merged**